src/components/implementation/tests/unit_defcompinfo/unit_defcompinfo.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2016, Phani Gadepalli and Gabriel Parmer, GWU, gparmer@gwu.edu.
+ *
+ * This uses a two clause BSD License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <cos_component.h>
+#include <cobj_format.h>
+#include <cos_defcompinfo_api.h>
+
+#undef assert
+#define assert(node) do { if (unlikely(!(node))) { debug_print("assert error in @ "); *((int *)0) = 0; } } while (0)
+#define PRINT_FN prints
+#define debug_print(str) (PRINT_FN(str __FILE__ ":" STR(__LINE__) ".\n"))
+#define BUG() do { debug_print("BUG @ "); *((int *)0) = 0; } while (0);
+#define SPIN() do { while (1) ; } while (0)
+
+#define CHILD_COMP_COUNT   2
+#define CHILD_UNTYPED_SIZE (1<<24)
+#define CHILD_SCHED_ID     0
+#define CHILD_SCHED_CYCS   TCAP_RES_INF 
+#define CHILD_SCHED_PRIO   TCAP_PRIO_MAX
+
+static void
+cos_llprint(char *s, int len)
+{ call_cap(PRINT_CAP_TEMP, (int)s, len, 0, 0); }
+
+int
+prints(char *s)
+{
+	int len = strlen(s);
+
+	cos_llprint(s, len);
+
+	return len;
+}
+
+int __attribute__((format(printf,1,2)))
+printc(char *fmt, ...)
+{
+	  char s[128];
+	  va_list arg_ptr;
+	  int ret, len = 128;
+
+	  va_start(arg_ptr, fmt);
+	  ret = vsnprintf(s, len, fmt, arg_ptr);
+	  va_end(arg_ptr);
+	  cos_llprint(s, ret);
+
+	  return ret;
+}
+
+int is_booter = 1;
+extern vaddr_t cos_upcall_entry;
+struct cos_defcompinfo child_defci[CHILD_COMP_COUNT];
```

Comment:
I used `defci` for `defcompinfo` in the variable name all throughout, like `ci` for `compinfo`.